lib/kube/grpc/utils.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// getWebAddr returns the address of the web server that is running on the
+// proxy. This is used to dial to Kube proxy and to generate the SNI value for
+// the TLS connection. Since the grpc server is only enabled if the proxy is
+// listening with multiplexing enabled, this function will always be able to
+// return a valid address.
+func getWebAddr(proxyAddr string) (string, string, error) {
```

Comment:
Since there are two strings being returned it might be nice to use named return types to differentiate between them or be more explicit in the docs about what is being returned possibly with an example.